### PR TITLE
Added processing of unknown subdivision targeting regions.

### PR DESCRIPTION
**Pull request by aseren**
Change `Regional ad relevance` value to `Disable` if geo-target endpoint returns unknown region.

Resolves https://github.com/brave/brave-browser/issues/25342

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Change locale to United States
- Override geo-target endpoint response with `{"country":"US", "region":"NO REGION"}`
- Start browser with ads enabled
- Check that `Regional ad relevance` value is set to `Disable`

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/subdivision_targeting.cc [hunk 1]
```diff
@@ -139,22 +145,21 @@ void SubdivisionTargeting::MaybeAllowForLocale(const std::string& locale) {
 
   const std::string& subdivision_code = GetSubdivisionCode();
   if (subdivision_codes.find(subdivision_code) == subdivision_codes.cend()) {
-    AdsClientHelper::GetInstance()->SetBooleanPref(
-        prefs::kShouldAllowSubdivisionTargeting, false);
-    MaybeResetSubdivisionCodeToAutoDetect();
-    return;
+    BLOG(1, "Unknown subdivision target " << subdivision_code << " for "
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/subdivision_targeting.cc:148]: nit: change `target` to `code`

File: vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/subdivision_targeting_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  MockUrlResponses(ads_client_mock_, url_responses);
+
+  // Act
+  subdivision_targeting_->MaybeFetch();
+
+  // Assert
+  EXPECT_TRUE(subdivision_targeting_->ShouldAllow());
+  EXPECT_FALSE(subdivision_targeting_->IsDisabled());
+
+  EXPECT_EQ(AdsClientHelper::GetInstance()->GetStringPref(
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/subdivision_targeting_unittest.cc:50]: nit: expected value on the `lhs`

File: vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/subdivision_targeting_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+namespace ads {
+
+class BatAdsSubdivisionTargetingTest : public UnitTestBase {
+ protected:
+  BatAdsSubdivisionTargetingTest() = default;
+
+  void SetUp() override {
+    UnitTestBase::SetUp();
+
+    MockLocaleHelper(locale_helper_mock_, "en-US");
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/subdivision_targeting_unittest.cc:26]: nit: move to `SetUpMocks`

File: vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/subdivision_targeting_unittest.cc [hunk 4]
```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/geographic/subdivision/subdivision_targeting.h"
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/geographic/subdivision/subdivision_targeting_unittest.cc:6]: Could we add tests for when the server responds with invalid JSON and empty country and region

**Review by tmancey — APPROVED** [review]
LGTM++ (please see nits)
